### PR TITLE
[operator-trivy] The new module

### PR DESCRIPTION
**Pull request by zuzzas**
## Description

Added trivy-operator and a dashboard.

## Why do we need it, and what problem does it solve?

To scan image vulnerabilities.

## What is the expected result?

Image vulnerabilities are scanned and reported to Prometheus.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: operator-trivy
type: feature
summary: The new module.
impact_level: default
```

### CODE REVIEW

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clustercompliancereports.yaml [hunk 1]
```diff
@@ -0,0 +1,257 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_clustercompliancereports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clusterconfigauditreports.yaml [hunk 2]
```diff
@@ -0,0 +1,173 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_clusterconfigauditreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clusterinfraassessmentreports.yaml [hunk 3]
```diff
@@ -0,0 +1,172 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_clusterinfraassessmentreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clusterrbacassessmentreports.yaml [hunk 4]
```diff
@@ -0,0 +1,172 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_clusterrbacassessmentreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_configauditreports.yaml [hunk 5]
```diff
@@ -0,0 +1,174 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_configauditreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_exposedsecretreports.yaml [hunk 6]
```diff
@@ -0,0 +1,206 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_exposedsecretreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_infraassessmentreports.yaml [hunk 7]
```diff
@@ -0,0 +1,173 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_infraassessmentreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_rbacassessmentreports.yaml [hunk 8]
```diff
@@ -0,0 +1,173 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_rbacassessmentreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_vulnerabilityreports.yaml [hunk 9]
```diff
@@ -0,0 +1,258 @@
+---
```
Comment (RomanenkoDenys) [inline modules/500-operator-trivy/crds/aquasecurity.github.io_vulnerabilityreports.yaml:1]: Please add a comment where we can download the new version of the crd.

File: modules/values-default.yaml [hunk 10]
```diff
@@ -27,6 +27,7 @@ namespaceConfiguratorEnabled: true
 nodeManagerEnabled: true
 operatorPrometheusCrdEnabled: true
 operatorPrometheusEnabled: true
+operatorTrivyEnabled: true
```
Comment (RomanenkoDenys) [inline modules/values-default.yaml:30]: NIT I think trivy operator should be false by default

File: modules/values-managed.yaml [hunk 11]
```diff
@@ -20,6 +20,7 @@ monitoringPingEnabled: true
 namespaceConfiguratorEnabled: true
 operatorPrometheusCrdEnabled: true
 operatorPrometheusEnabled: true
+operatorTrivyEnabled: true
```
Comment (RomanenkoDenys) [inline modules/values-managed.yaml:23]: NIT I think trivy operator should be false by default

**Review by RomanenkoDenys — COMMENTED** [review]
Looks good, have some NITS

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clustercompliancereports.yaml (hunk 1 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_clustercompliancereports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clusterconfigauditreports.yaml (hunk 2 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_clusterconfigauditreports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clusterinfraassessmentreports.yaml (hunk 3 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_clusterinfraassessmentreports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_clusterrbacassessmentreports.yaml (hunk 4 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_clusterrbacassessmentreports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_configauditreports.yaml (hunk 5 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_configauditreports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_exposedsecretreports.yaml (hunk 6 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_exposedsecretreports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_infraassessmentreports.yaml (hunk 7 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_infraassessmentreports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_rbacassessmentreports.yaml (hunk 8 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_rbacassessmentreports.yaml:1]: Done.

File: modules/500-operator-trivy/crds/aquasecurity.github.io_vulnerabilityreports.yaml (hunk 9 above)
Comment (zuzzas) [reply modules/500-operator-trivy/crds/aquasecurity.github.io_vulnerabilityreports.yaml:1]: Done.

File: modules/values-default.yaml (hunk 10 above)
Comment (zuzzas) [reply modules/values-default.yaml:30]: Done.

**Review by RomanenkoDenys — COMMENTED** [review]
LGTM

**Review by RomanenkoDenys — DISMISSED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by RomanenkoDenys — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

File: .github/workflows/trivy-db-schedule.yml [hunk 12]
```diff
@@ -0,0 +1,63 @@
+#
+# THIS FILE IS GENERATED, PLEASE DO NOT EDIT.
```
Comment (diafour) [inline .github/workflows/trivy-db-schedule.yml:2]: Corresponding workflow template is forgotten.

**Review by kvaps — APPROVED** [review]
linstor webhook ok

File: modules/500-operator-trivy/charts/helm_lib [hunk 13]
```diff
@@ -0,0 +1 @@
+/deckhouse/helm_lib
```
Comment (yalosev) [inline modules/500-operator-trivy/charts/helm_lib:1]: Wrong symlink. Should be `/deckhouse/helm_lib/charts/deckhouse_lib_helm`

File: modules/500-operator-trivy/templates/pdb.yaml [hunk 14]
```diff
@@ -0,0 +1,15 @@
+---
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
```
Comment (yalosev) [inline modules/500-operator-trivy/templates/pdb.yaml:6]: do we really need the hook annotation here? PDB works well without it but this annotation can create some stupid problems on update

**Review by yalosev — CHANGES_REQUESTED** [review]
A few questions

File: modules/500-operator-trivy/templates/pdb.yaml (hunk 14 above)
Comment (zuzzas) [reply modules/500-operator-trivy/templates/pdb.yaml:6]: Honestly, I've just copied it from another PDB in the codebase. I've assumed that it's some forbidden magical incantation to please Helm.

Comment (zuzzas) [reply modules/500-operator-trivy/templates/pdb.yaml:6]: Fixed.

File: modules/500-operator-trivy/charts/helm_lib (hunk 13 above)
Comment (zuzzas) [reply modules/500-operator-trivy/charts/helm_lib]: Fixed.

File: .github/workflows/trivy-db-schedule.yml (hunk 12 above)
Comment (zuzzas) [reply .github/workflows/trivy-db-schedule.yml:2]: We've decided not to write any templates.
@shvgn

**Review by yalosev — DISMISSED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**